### ISSUE TITLE
query about convert_tacred_checkpoint 

### ISSUE THREAD

**Issue by aakashb95** [studio-ousia/luke#117]
Hey,
I am using the files from the TACRED's json directory to finetune LUKE using your repo and script.

Trained on 40GB A100 GPU.
Training args :
```
python -m examples.cli \
--model-file=../luke-model-reqs/luke_large_500k.tar.gz \
--output-dir=../tacred_finetuned \
relation-classification run \
--data-dir=../tacred/json/ \
--train-batch-size=4 \
--gradient-accumulation-steps=8 \
--learning-rate=1e-5 \
--num-train-epochs=5
```

metrics after finetuning:
```
[2022-03-16 20:25:26,707] [INFO] Results: {
  "best_epoch": 3,
  "dev_f1": 0.7180689415646434,
  "dev_f1_epoch0": 0.6738297673549472,
  "dev_f1_epoch1": 0.6909220617085786,
  "dev_f1_epoch2": 0.7050176934942383,
  "dev_f1_epoch3": 0.7180689415646434,
  "dev_f1_epoch4": 0.7161278925984808,
  "dev_precision": 0.6799868442690347,
  "dev_precision_epoch0": 0.6846402126447693,
  "dev_precision_epoch1": 0.6704742125302873,
  "dev_precision_epoch2": 0.6956132497761862,
  "dev_precision_epoch3": 0.6799868442690347,
  "dev_precision_epoch4": 0.6887529731566429,
  "dev_recall": 0.7606696100073583,
  "dev_recall_epoch0": 0.6633554083885209,
  "dev_recall_epoch1": 0.7126563649742458,
  "dev_recall_epoch2": 0.7146799116997793,
  "dev_recall_epoch3": 0.7606696100073583,
  "dev_recall_epoch4": 0.7457689477557027,
  "test_f1": 0.7124526100904053,
  "test_precision": 0.6914803283328615,
  "test_recall": 0.7347368421052631
} (run@main.py:119)
```

Then I used your [convert_tacred_checkpoint](https://gist.github.com/ikuyamada/2ef87bd24ca7d6c0581c6221caf190e0) gist
I had to comment out the assert statements.

convert tacred checkpoint args:
```
python convert_tacred_checkpoint.py \
--checkpoint_path=../tacred_finetuned/pytorch_model.bin \
--metadata_path=../luke-model-reqs/metadata.json \
--entity_vocab_path=../luke-model-reqs/entity_vocab.tsv \
--pytorch_dump_folder_path=../tacred_hf \
--model_size=large
```

When I use the hugging face checkpoint and test on a simple example, I get a false positive:
![image](https://user-images.githubusercontent.com/48802744/158661185-cae88346-f1c1-444c-bb33-c34cd9838417.png)

I have not modified TACRED or any code in your repo. Any suggestions on how to debug this?

**Comment by Ryou0634** [comment]
Hi, is the problem the model making an incorrect prediction?
I think that it is an expected behavior since neural models are by no means perfect.
They can be wrong with words or constructions that they haven't learned well.

If you want to improve the model or make the model work on specific types of examples, maybe you can add training data for fine-tuning.

**Comment by aakashb95** [comment]
> If the problem is that the transformer model shows an inconsistent behavior with the original model, please clarify that and feel free to open another issue : )

Yes, I feel it is inconsistent because, when I use the checkpoint on huggingface hub with the same example, I get the correct answer (per:employee_of) :
<img width="887" alt="image" src="https://user-images.githubusercontent.com/48802744/163556822-0dd55fb4-7263-49e0-8b0c-d32612a1f762.png">

My question is, I have used the exact same TACRED data and training arguments and got almost similar test_f1 like in the paper. Then I used convert_tacred_checkpoint script to convert the model to hugging face format and I get different results. (per:alternate_names)
How to debug this?

**Comment by Ryou0634** [comment]
I see.

There are two possibilities:
① The converted model is right and it just happens to make an incorrect prediction for that specific example.
② The converted model is somehow broken.

To confirm if ① is the case, you can calculate the F1 score on the test set with your converted model.
For that purpose, you may find [this script](https://github.com/studio-ousia/luke/blob/master/examples/relation_classification/evaluate_transformers_checkpoint.py) helpful.

For ②, you can check the weights between your trained model and converted model, for example, by inspecting `model.named_parameters()`. The names of the weights are slightly different, but the mapping should be straightforward.

**Comment by aakashb95** [comment]
The model fails across multiple examples that work in the default huggingface checkpoint. The weights also look quite different.
I will try a clean run again and come with doubts if any. Thanks for your help
